web/src/features/map/Map.tsx

```diff
@@ -153,14 +156,46 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
 
   useEffect(() => {
     // Run on first load to center the map on the user's location
-    if (!map || isError || !isFirstLoad) {
+    if (!map || isError || !isFirstLoad || !isSourceLoaded || !callerLocation) {
       return;
     }
-    if (data?.callerLocation && !selectedZoneId) {
-      map.flyTo({ center: [data.callerLocation[0], data.callerLocation[1]] });
+    if (callerLocation && !selectedZoneId) {
+      map.flyTo({ center: [callerLocation[0], callerLocation[1]] });
+
+      const handleIdle = () => {
+        if (map.isSourceLoaded(ZONE_SOURCE) && map.areTilesLoaded() && callerLocation) {
+          const source = map.getSource(ZONE_SOURCE);
+          const layer = map.getLayer('zones-clickable-layer');
+          if (!source) {
+            console.error(`Source "${ZONE_SOURCE}" not found`);
+            return;
+          }
+          if (!layer) {
+            console.error('Layer "zones-clickable-layer" not found or not rendered');
+            return;
+          }
+          const zoneFeature = getZoneIdFromLocation(map, callerLocation, ZONE_SOURCE);
+          if (zoneFeature) {
+            const zoneName = zoneFeature.properties.zoneId;
+            setUserLocation(zoneName);
+            console.log('User is in zone:', zoneName);
```

Comment:
Let's remove this logging

Reply:
Alternatively use debug logging behind an import.meta.dev if clause so it's removed at build.

Reply:
Had an unpushed commit with this removed but will take another look too